audio/AudioChannel.cc

```diff
@@ -202,6 +206,20 @@ namespace Pentagram {
 		} while (bytes != 0);
 	}
 
+	uint32 AudioChannel::getPlaybackLength() const {
+		if (!sample) {
+			return UINT32_MAX;
+		}
+		return ((sample->getPlaybackLength()) * 1000) / sample->getRate();
+	}
+
+	uint32 AudioChannel::getPlaybackPosition() const {
+		if (!sample) {
+			return UINT32_MAX;
+		}
+		return (overall_position * 1000) / sample->getRate();
```

Comment:
How likely would this be to overflow?

Reply:
It would probably overflow at the end of the ogg music tracks. More than 2 million samples isn't unexpected for music Will make this 64 bit

Reply:
Now fixed